scheduler/scheduler-api/src/main/java/org/ow2/proactive/scheduler/common/job/factories/model/DefaultModelJobValidatorFactory.java

```diff
@@ -0,0 +1,391 @@
+                TaskVariable duplicatedVariable = new TaskVariable(variable.getName(),
+                                                                   variable.getValue(),
+                                                                   line.trim(),
+                                                                   false);
+                checkVariableFormat(task, duplicatedVariable);
+                return;
+            }
+
+        } catch (MalformedURLException e) {
+            throw new JobCreationException(getDefaultErrorMessage(task, variable, model) +
+                                           " illegal url in " + MODEL_FROM_URL_MODEL + " model: " + e.getMessage(), e);
+        } catch (IOException e) {
+            throw new JobCreationException(getDefaultErrorMessage(task, variable, model) +
+                                           " cannot access url set in " + MODEL_FROM_URL_MODEL + " model.", e);
+        }
+    }
+
+    private <T extends Comparable<T>> void checkValueOf(Task task, TaskVariable variable, String model, T classObject)
+            throws JobCreationException {
+        String value = getValueAndCheckNull(task, variable, model);
+        convertTo(task, variable, value, model, classObject);
+    }
+
+    private <T extends Comparable<T>> void checkPrimitiveValueInRange(Task task, TaskVariable variable, String model,
+            String modelType, T classObject) throws JobCreationException {
+        if (!model.matches(modelType + "\\([^),]+,[^),]+\\)$")) {
+            throw new JobCreationException(getDefaultErrorMessage(task, variable, model) + " illegal " + modelType +
+                                           " expression in model");
+        }
+        String value = getValueAndCheckNull(task, variable, model);
+
+        String rangeString = model.substring(1 + modelType.length(), model.length() - 1);
+        String[] rangeArray = rangeString.split(",");
+        T min = convertTo(task, variable, rangeArray[0], model, classObject);
+        T max = convertTo(task, variable, rangeArray[1], model, classObject);
+
+        if (min.compareTo(max) > 0) {
+            throw new JobCreationException(getDefaultErrorMessage(task, variable, model) + " illegal range in model");
+        }
+
+        T target = convertTo(task, variable, value, model, classObject);
+
+        if ((min.compareTo(target) > 0) || target.compareTo(max) > 0) {
+            throw new JobCreationException(getDefaultErrorMessage(task, variable, model) + " value \"" + value +
+                                           "\" not in range (" + min + "," + max + ")");
+        }
+
+        if (rangeArray.length != 2) {
+            throw new JobCreationException(getDefaultErrorMessage(task, variable, model) + " illegal " + modelType +
+                                           " expression in model");
+        }
+
+    }
+
+    private <T extends Comparable<T>> T convertTo(Task task, TaskVariable variable, String value, String model,
```

Comment:
It looks really ugly and not efficient.

A clean and efficient OO approach would be to abstract the parsing and the validation in 2 different abstract classes: for instance `ModelConverter` and `ModelValidator`. Then, for each type, you define a concrete class which implements the behaviour.  For efficiency, you can store the instance of these converters and validators in a static Map.

This allows low coupling and strong cohesion.

An example is available in the following, you could even reuse it:

https://github.com/yfiton/yfiton/tree/master/yfiton-api/src/main/java/com/yfiton/api/parameter

You could also reuse valueOf for parsing value for primitives.